src/interactive/repl.ts

```diff
@@ -38,6 +39,65 @@ function startREPLCommand() {
 
     startREPL(false, true)
 }
+async function checkKillPersistentSession() {
+    if (persistentStopWarning || vscode.workspace.getConfiguration('julia').get<boolean>('persistentSession.stopWarning') === false) {
+        return true
+    }
+    else {
+        persistentStopWarning = true
+        try {
+            const agree = 'Yes'
+            const agreeAlways = 'Yes, always'
+            const disagree = 'No'
+            const choice = await vscode.window.showInformationMessage('This is a persistent tmux session. Do you want to close it?', agree, agreeAlways, disagree)
+            if (choice === disagree) {
+                return false
+            }
+            if (choice === agreeAlways) {
+                vscode.workspace.getConfiguration('julia').update('persistentSession.stopWarning', false, true)
+            }
+            if (choice === agree || choice === agreeAlways) {
+                return true
+            }
+        }
+        finally {
+            persistentStopWarning = false
+        }
+    }
+}
+async function stopREPLInPersistent() {
```

Comment:
I think this can live in `killREPL` directly. If not, let's call it `maybeKillPersistentSession`; that name seems a bit clearer.